jina/executors/__init__.py

```diff
@@ -536,3 +556,82 @@ def __call__(self, req_type, *args, **kwargs):
                     raise UnattachedDriver(d)
         else:
             raise NoDriverForRequest(req_type)
+
+
+class BaseFrameworkExecutor(BaseExecutor):
+    """
+    :class:`BaseFrameworkExecutor` is the base class for the executors using other frameworks internally, including
+        `tensorflow`, `pytorch`, `onnx`, and, `paddlepaddle`.
+
+    ..notes:
+        The derived classes must implement `build_model()` and `set_device()` methods.
+    """
+    def post_init(self):
+        super().post_init()
+        self.pre_set_device()
```

Comment:
if `pre_set_device` and `post_set_device` is an "either-or", then use a meta tag to branch the logics.
e.g. add a meta tag `gpu_placement`: `before`/`after`